src/core/server/models/seenComments/seenComments.ts

```diff
@@ -84,9 +84,21 @@ export async function markSeenComments(
   storyID: string,
   userID: string,
   commentIDs: string[],
-  now: Date
+  now: Date,
+  markAllAsSeen?: boolean
 ) {
-  const comments = reduceCommentIDs(commentIDs, now);
+  let comments;
+  if (markAllAsSeen) {
+    const markAllCommentIDs = (
+      await mongo
+        .comments()
+        .find({ storyID }, { projection: { id: 1 } })
```

Comment:
This needs to filter by `tenantID` as well. It's a small change, so I made it and pushed it up.